resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/CharsetManager.java

```diff
@@ -369,23 +369,25 @@ private String internalGetCharsetFor(Preferences prefs, Preferences derivedPrefs
 		String charset = null;
 
 		// try to find the encoding in regular and then derived preferences
+		String key1 = getKeyFor(resourcePath);
 		if (prefs != null)
-			charset = prefs.get(getKeyFor(resourcePath), null);
+			charset = prefs.get(key1, null);
 		// derivedPrefs may be not null, only if #isDerivedEncodingStoredSeparately returns true
 		// so the explicit check against #isDerivedEncodingStoredSeparately is not required
 		if (charset == null && derivedPrefs != null)
-			charset = derivedPrefs.get(getKeyFor(resourcePath), null);
+			charset = derivedPrefs.get(key1, null);
```

Comment:
Maybe better put into a method (372-378) then one can use the same call and don't need "key1" and "key2" ...

Reply:
done